### ISSUE TITLE
Ability to ignore branches repeated only by `else` in `Lint/DuplicateBranch`

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#13146]
I would like to propose a new configuration option (or do this by default) to ignore repeated branches when we have some branch which is repeated only in `else`. Sometimes it is better (more descriptive) to be explicit by listing all the conditions/options and fallback to `else`.

```ruby
# good
case option
when "bar"
  do_something
when "baz"
  do_something_else
else
  do_something
end

# bad, the same is repeated in both branches
if condition
  do_something
else
  do_something
end
```

**Comment by Earlopain** [comment]
+1 from me. It's very awkward to rewrite into something that doesn't trigger RuboCop.

Most times merging conditions is straightforward but if I want to do the same for `else` I would need to check all conditions beforehand, duplicating a bunch of stuff. So I just end up disabling the cop for these instances.

**Comment by fatkodima** [comment]
@bbatsov Wdyt? If ok, I would like to implement it.
